azkaban-common/src/main/java/azkaban/executor/FetchActiveFlowDao.java

```diff
@@ -56,9 +59,9 @@ public FetchActiveFlowDao(final DatabaseOperator dbOperator) {
     // Select running and executor assigned flows
     private static final String FETCH_ACTIVE_EXECUTABLE_FLOW =
         "SELECT ex.exec_id exec_id, ex.enc_type enc_type, ex.flow_data flow_data, et.host host, "
-            + "et.port port, et.id executorId, et.active executorStatus"
+            + "et.port port, ex.executor_id executorId, et.active executorStatus"
             + " FROM execution_flows ex"
-            + " INNER JOIN "
+            + " LEFT JOIN "
```

Comment:
why is this needed?  can you give an example of wrong case which might be caused by original sql

Reply:
Hi @juhoautio, I have two questions regarding this change:
1. `LEFT JOIN` will also include executions that are in `PREPARING` status and haven't been dispatched and assigned an executor yet. When a web server is restarted, it will call `loadRunningExecutions` and `loadQueuedFlows`. The flows in `PREPARING` status will be included in both `runningExecutions` and `queuedFlows`. They will be finalized in `updateExecutions` because the executor is null. They will again be dispatched by `queueProcessor` as well. We've noticed one incident from our current web server log. Although it doesn't impact users too much, I think it's still an issue since the `runningExecutions` definition doesn't quite match what is fetched from DB.
2. For those executions that have been assigned an executor id but the `executor_id` is removed from executor table, this query will return the `executor_id` from `execution_flows` table, which is not null. Can the execution still be finalized in `updateExecutions`?

Reply:
@juhoautio Thanks, the solution sounds good to me.

Reply:
@jamiesjc were you expecting me to make a PR for this? If yes, let me know and I'll do it.

Reply:
Sure, please go ahead with the fix. I'll review and merge it.